controllers/datamover.go

```diff
@@ -186,3 +203,187 @@ func (r *DPAReconciler) getDataMoverLabels() map[string]string {
 	labels[oadpv1alpha1.DataMoverDeploymentLabel] = "True"
 	return labels
 }
+
+// Check if there is a valid user supplied restic secret
+func (r *DPAReconciler) validateDataMoverCredential(resticsecret *corev1.Secret) bool {
+	if resticsecret == nil {
+		return false
+	}
+	for key, val := range resticsecret.Data {
```

Comment:
We want to update this to check if the key exists at all, right now I get no error if the key isn't set